core/amber/src/main/scala/edu/uci/ics/texera/web/service/WorkflowService.scala

```diff
@@ -153,12 +173,21 @@ class WorkflowService(
         )
       )
     }
+
     var executionID: Long = -1 // for every new execution,
+
     // reset it so that the value doesn't carry over across executions
     if (WorkflowService.userSystemEnabled) {
-      executionID = ExecutionsMetadataPersistService.insertNewExecution(wId, uidOpt)
+      executionID = ExecutionsMetadataPersistService.insertNewExecution(wId, vId, uidOpt)
     }
-    new WorkflowContext(jobID, uidOpt, wId, executionID)
+    new WorkflowContext(
+      jobID,
+      uidOpt,
+      getLatestVersion(UInteger.valueOf(wId)).intValue(),
```

Comment:
just reuse vid from line above since in line 116 you already got latest version and assigned it to vid